### ISSUE TITLE
LNK2019	unresolved external symbol error when using RegistrationTools

### ISSUE THREAD

**Issue by a1ibekov** [CloudCompare/CloudCompare#2047]
**Describe the bug**

I am trying to create a custom plugin on top of "Example Plugin" project. I want to use CloudCompare fine registration by using "ccRegistrationTools". I was able to add the header file (#include "ccRegistrationTools.h"), but when i tried to use the ICP function itself 
i get the following error: 
-  LNK2019	unresolved external symbol "public: static bool __cdecl ccRegistrationTools::ICP(class ccHObject *,class ccHObject *,class ccGLMatrix &,double &,double &,unsigned int &,struct CCCoreLib::ICPRegistrationTools::Parameters const &,bool,bool,class QWidget *)" (?ICP@ccRegistrationTools@@SA_NPEAVccHObject@@0AEAVccGLMatrix@@AEAN2AEAIAEBUParameters@ICPRegistrationTools@CCCoreLib@@_N5PEAVQWidget@@@Z) referenced in function "public: __cdecl <lambda_15b6b6ee015ae46678748843612539d9>::operator()(void)const " (??R<lambda_15b6b6ee015ae46678748843612539d9>@@QEBA@XZ)	

The QCC_DB_LIB project which contains "ccRegistrationTools.h and .cpp" files and CCCoreLib that has "RegistrationTools.h and .cpp" files were build without problems. 

**Additional context**

Here is my linker settings: 
![image](https://github.com/user-attachments/assets/e24f9339-8ce8-4e92-8ffb-0eda191fb23f)

**Your environment**
I have:
- Qt version 5.15.2
- CloudCompare version v.2.14 alpha 
- Visual Studio 17 2022

**Comment by dgirardeau** [comment]
Hi. You probably have to include the cpp file as well.

**Comment by a1ibekov** [comment]
> Hi. You probably have to include the cpp file as well.

Hi, I attached it here 
[issue.zip](https://github.com/user-attachments/files/16467769/issue.zip)

**Comment by dgirardeau** [comment]
I mean you have to add the cpp file to your project/plugin. Via its CMake configuration file.

**Comment by a1ibekov** [comment]
> I mean you have to add the cpp file to your project/plugin. Via its CMake configuration file.

but as i understood, i should be able to use all the built in functions of CloudCompare without adding the cpp files, right? So maybe I am doing something wrong

**Comment by dgirardeau** [comment]
Nope, inside a plugin you only have access to what is exposed in the ccMainAppInterface: https://github.com/CloudCompare/CloudCompare/blob/master/libs/CCPluginAPI/include/ccMainAppInterface.h

If you want to reuse a tool or a dialog, you'll have to 'duplicate' it (but no need to copy it, you just need to add the class in your plugin).

**Comment by a1ibekov** [comment]
Hmm, interesting. So everything that belongs to  "CCCoreLib::" I can use without adding the files itself, because it worked with "CCCoreLib::DistanceComputationTools::computeCloud2CloudDistances". However, if I want to use methods from qCC folder I should I add them. Is it correct?

**Comment by a1ibekov** [comment]
I understood. Thank you so much! I added the file and updated the code. But when I test it in CloudCompare the final transformation is bad and doesn't align point clouds properly. In Console window, there is another transformation that is printed by [GraphicalTransformationTool] and when I apply these values I get perfect alignment. 
![image](https://github.com/user-attachments/assets/bc648738-33d0-4778-94e9-5addeb1aba79)

**Comment by a1ibekov** [comment]
Got it, thank you so much! I added the file and updated code to run registration. However, when I test it in CloudCompare the alignment is bad. I moved one point cloud using "Translate/Rotate" tool which printed the transformation from "GraphicalTransformationTool". We can use it as the ground truth value. When I print the output "ccGLMatrix" as"My registration" I get values that are quite different: 
![image](https://github.com/user-attachments/assets/da15f1c9-12bd-472b-8ec8-bd1cbea7eba6)
And here is the result of using "fine alignment" built it plugin of CloudCompare: 
![image](https://github.com/user-attachments/assets/58020712-ff32-4e63-aaac-bee3b834304e)

Parameters that I am using: 
![image](https://github.com/user-attachments/assets/8db9a411-0fa3-4186-af99-8baedda430db)

**Comment by dgirardeau** [comment]
Are you sure about your final overlap ratio? Shouldn't it be 1.0? (instead of 0.01, which would correspond to 1%)

**Comment by a1ibekov** [comment]
Yes, I think that was the problem. I realized that depending on the case, we will need to tune this parameter. For example, when there are partial scans, we shouldn't set it to 100% as it might incorrectly "over" align point clouds. Thank you so much for your help and quick responses!
